### ISSUE TITLE
should we force `chi2xspecvar` users to only subtract, not fit, the background?

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1543]
This is a follow-up from working on #356 

If 

1. the user is fitting a model to a PHA dataset rather than subtracting it,
2. a source or background group has 0 counts in it, 
3. the user is using `chi2xspecvar`, 
4. and the PHA dataset did not come with it's own errors (and so we are calculating them)
 
then we **CAN NOT MATCH XSPEC**. This is because XSPEC special cases the error in this situation (see #356) and it is **NOT_AT_ALL_OBVIOUS** how we can take that approach (which combines the source and background cases) and apply it to the modelling case (which treats the source and background cases separately.

So, what should happen to a user in this case? I can see

a) warning the user
b) refusing to do anything (i.e. error out)
c) silently do something, but it's not really obvious what it has done

Actually, given that XSPEC doesn't handle item 1, we perhaps should just never let a user fit a background model when using `chi2xspecvar`.

**Comment by anetasie** [comment]
@DougBurke I agree, If the background is present then the user should not be using this statistics without background subtraction, so we should issue a warning. I would not prevent user from doing a fit if they want. Warning would be good.

**Comment by DougBurke** [comment]
@anetasie I'm not sure of the priority on this - unless it's given a high priority I'm not sure I'll get around to it (the change isn't conceptually hard, but it actually is hard to work out where to trigger the warning and how best to do it)

**Comment by anetasie** [comment]
I would not give this a very high priority. Perhaps we should make a comment in the documentation for this statistics that it should be applied to the background subtracted data only and caveat about the agreement with XSPEC values.
